filter/filter.go

```diff
@@ -422,27 +422,78 @@ func (c *Client) GetDimensionOptionsBatchProcess(ctx context.Context, userAuthTo
 	return eTag, batch.ProcessInConcurrentBatches(batchGetter, batchProcessor, batchSize, maxWorkers)
 }
 
+// CreateFlexibleBlueprint creates a flexible filter blueprint and returns the associated filterID and eTag
+func (c *Client) CreateFlexibleBlueprint(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version string, names []string, population_type string) (filterID, eTag string, err error) {
+	ver, err := strconv.Atoi(version)
+	if err != nil {
+		return "", "", err
+	}
+
+	dimensions := make([]ModelDimension, len(names))
+	for i, name := range names {
+		dimensions[i] = ModelDimension{Name: name}
+	}
+
+	cb := createFlexBlueprintRequest{
+		Dimensions:     dimensions,
+		Dataset:        Dataset{DatasetID: datasetID, Edition: edition, Version: ver},
+		PopulationType: population_type,
+	}
+
+	reqBody, err := json.Marshal(cb)
+	if err != nil {
+		return "", "", err
+	}
+
+	respBody, eTag, err := c.postBlueprint(ctx, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version, reqBody)
+	if err != nil {
+		return "", "", err
+	}
+
+	var respData createFlexBlueprintResponse
+	if err = json.Unmarshal(respBody, &respData); err != nil {
+		return "", "", err
+	}
+
+	return respData.FilterID, eTag, nil
+}
+
 // CreateBlueprint creates a filter blueprint and returns the associated filterID and eTag
 func (c *Client) CreateBlueprint(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version string, names []string) (filterID, eTag string, err error) {
 	ver, err := strconv.Atoi(version)
 	if err != nil {
 		return "", "", err
 	}
 
-	cb := CreateBlueprint{Dataset: Dataset{DatasetID: datasetID, Edition: edition, Version: ver}}
+	dimensions := make([]ModelDimension, len(names))
+	for i, name := range names {
+		dimensions[i] = ModelDimension{Name: name}
+	}
 
-	var dimensions []ModelDimension
-	for _, name := range names {
-		dimensions = append(dimensions, ModelDimension{Name: name})
+	cb := createBlueprint{
+		Dimensions: dimensions,
+		Dataset:    Dataset{DatasetID: datasetID, Edition: edition, Version: ver},
 	}
 
-	cb.Dimensions = dimensions
+	reqBody, err := json.Marshal(cb)
+	if err != nil {
+		return "", "", err
+	}
 
-	b, err := json.Marshal(cb)
+	respBody, eTag, err := c.postBlueprint(ctx, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version, reqBody)
 	if err != nil {
 		return "", "", err
 	}
 
+	if err = json.Unmarshal(respBody, &cb); err != nil {
+		return "", "", err
+	}
+
+	return cb.FilterID, eTag, nil
+}
+
+func (c *Client) postBlueprint(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version string, reqBody []byte) ([]byte, string, error) {
```

Comment:
this function is behaviour required for POSTing to the `/filters` endpoint common to both the above methods